src/tuntap/windows/c_tuntap_windows.cpp

```diff
@@ -101,13 +104,13 @@ size_t c_tuntap_windows_obj::read_from_tun_separated_addresses(
 		std::array<unsigned char, 14 + 40 + 65535> input_buffer; // eth header + ipv6 header + max ipv6 payload
 		size_t readed_bytes = read_from_tun(input_buffer.data(), input_buffer.size());
 		if (readed_bytes == 0) return 0;
-		auto ipv6_header_begin = input_buffer.begin() + 14;
+		auto ipv6_header_begin = input_buffer.begin();
 		// copy addresses
 		std::copy(ipv6_header_begin + 8, ipv6_header_begin + 24, src_binary_address.begin());
 		std::copy(ipv6_header_begin + 24, ipv6_header_begin + 40, dst_binary_address.begin());
 		// copy content without addresses
 		std::copy(ipv6_header_begin, ipv6_header_begin + 8, data); // before addresses
-		size_t ipv6_payload_size = readed_bytes - 14 - 40; // size of received bytes without eth and ipv6 header
+		size_t ipv6_payload_size = readed_bytes - 40; // size of received bytes without ipv6 header
 		std::copy(ipv6_header_begin + 40, ipv6_header_begin + 40 + ipv6_payload_size, data + 8);
 		return readed_bytes - src_binary_address.size() - dst_binary_address.size();
 }
```

Comment:
OVERALL:

we want to avoid:

- using x.begin()+n or x.front()+n , please grep for begin and front overall
- raw std::copy, in addition of going out of bounds, there can be other invalid behaviour if ranges would happen to overlap.  Could be marked at calling function with comment: 
stdplus::copy......(....) // [[optimize]] if we are sure that the ranges are correct (and not-overlapping), and if this becomes important for performance, this could be replaced with some less secure copy... function